src/org.hdfgroup.hdfview/hdf/view/dialog/NewStringAttributeDialog.java

```diff
@@ -542,7 +542,24 @@ else if (tsize == 8) {
             }
 
             if (tclass == Datatype.CLASS_FLOAT) {
-                if ((tsize == 4) || (tsize == -1)) {
+                if ((tsize == 2)) {
+                    float[] f = new float[arraySize];
```

Comment:
Shouldn't this use a `short[]` rather than a `float[]` since the DataProviderFactory implementation for this appears to be trying to parse a `short` value and then convert it to a `float` in `getDataValue`?

Reply:
This dialog is not used for HDF5 types